rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,233 @@
+
+### Notification Requirements
+For a notification to be actionable by analysis, it must fulfill specific requirements. Specifically, the notification must include information about the bundles/data that belong to the same sequencing set and potentially need to be co-processed.
+
+We propose the term **Data Aggregation and Processing Set (DAPS)** to encompass all the different alternative representations of such sets. The only requirement imposed on such these representations is to be reproducible and resolvable to an immutable sets of data. Example representations of DAPS can be:
+
+*   A listing of all relevant bundles along with versions
+*   A search query accompanied by a timestamp (to guarantee reproducibility and immutability)
+*   Other future representations as project needs evolve (e.g. linking to a particular project bundle version)
+
+Critically, the submitted DAPS will not explicitly dictate the files that have to be co-processed in pipeline instances as this would require specialized knowledge of the pipelines by the submitter. Instead, **a DAPS indicates that aggregation of data must be performed only from the list of data provided (submission envelope), but the actual aggregation is delegated to the process of analysis initiation, subject to well defined modality-specific metadata requirements**. 
+
+The imposition of specific requirements to metadata is part of this RFC proposal and a key component of the proposed DAPS solution.
+
+This approach provides a generalized signaling mechanism suitable for all foreseeable data types while accounting for data type-specific requirements.
+
+The DAPS approach has common elements with project-level bundles that have been proposed in other contexts, but differs in the following ways:
+
+1. It does not impose any specific aggregation level on the data allowing for more flexibility and accommodating future unforeseen processing scenarios
+2. Allows specification of bundles included in a DAPS in many alternative different ways (listing, search, etc…)
+3. Does not impose the creation of a new bundle type that is tied to a specific data modality
+4. Dictates specific metadata requirements that guarantee the correct processing of all foreseeable data modalities
+5. Decouples analysis initiation from data representation allowing for future flexibility
+
+## General Metadata Requirements
+In the above framework, analysis pipelines will receive a set of bundles that _potentially_ need to be co-processed. As aforementioned, this signal will not explicitly identify the bundles that need to be co-processed in a single pipeline, instead, it will serve as a notification **indicating that any co-processing is to be done only within the boundaries of this set of files (submission envelope). **
+
+Analysis can initiate multiple pipeline invocations as a result of a single DAPS being submitted.
```

Comment:
How and why would you need to /want to?

Reply:
The goal is to not have one to one restriction on analysis to primary data.  Multiple algorithms or types of analysis are allowed.

why would we want to restrict the systems in this way?

Reply:
I understand not wanting to unduly restrict the system at the same time I also don't want to overcomplicate the system to enable a rare case which can be solved using more simple means

Reply:
I don't see any complexity here. If one pipeline can be invoked then it's trivial to invoke multiple. Maybe not even worth saying.